aplib.net-demo/Assets/Scripts/MainMenu/MenuButtons.cs

```diff
@@ -1,48 +1,120 @@
+    /// </summary>
+    public bool isOnGameSettings = false;
+
+    /// <summary>
+    /// This bool communicates with the mouse lock script to enable and disable the curser.
+    /// </summary>
+    public bool IsCursurNeeded => isOnGameSettings;
+
+    /// <summary>
+    /// This string keeps track of what scene we are in.
+    /// </summary>
+    private string _currentSceneName = "";
+
+    /// <summary>
+    /// Name of the start screen.
+    /// </summary>
+    [SerializeField]
+    private string _sceneNameStartingMenu = "Settings"; // TODO:: Load main start screen
+
+    /// <summary>
+    /// Name of the game scene
+    /// </summary>
+    [SerializeField]
+    private string _sceneNameGame = "MeleeWeaponMenu"; // TODO:: Load main game screen
+
+    // Looks at which canvas is needed by checking the current scene.
     private void Start()
     {
-        menuCanvas.SetActive(true);
-        isOnSettings = false;
-        settingCanvas.SetActive(false);
+        _currentSceneName = SceneManager.GetActiveScene().name;
+
+        if (_currentSceneName == _sceneNameStartingMenu) // Are we at the starting screen?
+        {
+            menuCanvas.SetActive(true);
+            settingMenuCanvas.SetActive(false);
+            settingGameCanvas.SetActive(false);
+            isOnMenuSettings = false;
+            isOnGameSettings = false;
+        }
+        else if (_currentSceneName == _sceneNameGame) // Are we at the main gaming scene?
+        {
+            menuCanvas.SetActive(false);
+            settingMenuCanvas.SetActive(false);
+            settingGameCanvas.SetActive(false);
+            isOnMenuSettings = false;
+            isOnGameSettings = false;
+        }
+        else // TODO:: Remove when game is done. This is for future ease.
+        {
+            Debug.Log("Scene names are wrong. Check MenuButtons script");
+        }
     }
 
     /// <summary>
     /// When the play button in the main menu is clicked, it wil teleport the player to the game.
-    /// Right now it is the melee weapon scene to prove concept.
     /// </summary>
     public void PlayGame()
     {
-        // TODO:: Load main game scene
-        SceneManager.LoadScene("MeleeWeapon");
+        SceneManager.LoadScene(_sceneNameGame);
     }
 
     /// <summary>
-    /// This button toggles the menu canvas off and the setting canvas on.
+    /// When the Quit To Menu button in the game setting is clicked, it will take you back to the menu.
     /// </summary>
-    public void SettingToggle()
+    public void ToMenu()
     {
-        isOnSettings = !isOnSettings;
-        settingCanvas.SetActive(isOnSettings);
-        menuCanvas.SetActive(!isOnSettings);
+        SceneManager.LoadScene(_sceneNameStartingMenu);
+    }
+
+    /// <summary>
+    /// This button toggles the menu canvas off and the setting canvas on or game settings on and off.
+    /// </summary>
+    public void ToggleSettings()
```

Comment:
The "resume" button does not immediately take you back to the game, the mouse is still visible when you click it as if you are in a menu

Reply:
Is it not smart to have the bug ticket fix this instead of this ticket?